### PR TITLE
Enable/Disable Filters & Cropping Modal

### PR DESCRIPTION
**Pull request by matthewrh**
- Added button for showing/hiding filters
- Added button for opening cropping modal
- Moved cropping window and button to modal

Screen Shot 2019-11-25 at 1 11 42 PM

Screen Shot 2019-11-25 at 1 11 55 PM

Screen Shot 2019-11-25 at 1 12 11 PM

### CODE REVIEW

**Comment by matthewrh** [comment]
Needed to implement changes from #7 - That PR is now redundant.

File: src/pages/Post.js [hunk 1]
```diff
@@ -517,6 +522,39 @@ class PostPage extends React.Component {
...
+        filterToggleColor: "#a32512"
+      })
+    } else {
+      document.getElementById("yearBoxes").style.display = "none"
+      document.getElementById("schoolBoxes").style.display = "none"
+      this.setState({
+        filterToggle: false,
+        filterToggleText: "Add Filters",
+        filterToggleColor: "#12a340"
+      })
```
Comment (joshdoman) [inline src/pages/Post.js:542]: Good but we will want to refactor this. First, it's generally not good practice in React to use ```document.getElementById```. Instead, we should be using React refs. Here is documention on that: https://reactjs.org/docs/refs-and-the-dom.html. It is also not necessary to set state for the text or the color. Instead, we should check ```filterToggle``` in ```render()``` and set the text and color accordingly.

File: src/pages/Post.js [hunk 2]
```diff
@@ -547,30 +585,47 @@ class PostPage extends React.Component {
...
-                      <input className="is-checkradio is-small" id="WH" type="checkbox" checked={this.state.filters.school.WH} name="school_WH" onClick={this.setCheckBoxState} />
+                      <input className="is-checkradio is-small" id="WH" type="checkbox" checked={this.state.filters.school.WH} name="school_WH" onClick={this.setCheckBoxState}/>
                       <label for="WH">Wharton</label>
                       <input className="is-checkradio is-small" id="EAS" type="checkbox" checked={this.state.filters.school.EAS} name="school_EAS" onClick={this.setCheckBoxState}/>
                       <label for="EAS">SEAS</label>
-                      <input className="is-checkradio is-small" id="NUR" type="checkbox" checked={this.state.filters.school.NUR} name="school_NUR" onClick={this.setCheckBoxState} />
+                      <input className="is-checkradio is-small" id="NUR" type="checkbox" checked={this.state.filters.school.NUR} name="school_NUR" onClick={this.setCheckBoxState}/>
                       <label for="NUR">Nursing</label>
                     </div>
                   </div>
```
Comment (joshdoman) [inline src/pages/Post.js:631]: Let's refactor this into its own subcomponent.

File: src/pages/Post.js [hunk 3]
```diff
@@ -517,6 +522,39 @@ class PostPage extends React.Component {
...
+    }
+  }
+
+  openModal() {
+    this.setState({modalIsOpen: true})
+  }
+
+  afterOpenModal() {
+    this.setState({modalIsOpen: this.state.modalIsOpen})
+  }
```
Comment (joshdoman) [inline src/pages/Post.js:552]: What is the purpose of this? It doesn't seem to do anything.

File: src/pages/Post.js [hunk 4]
```diff
@@ -656,81 +711,61 @@ class PostPage extends React.Component {
...
+                              fontWeight: 500,
+                              fontSize: 18,
+                              color: "#ffffff"
+                            }}>
+                              Save Cropped Image
+                          </button>
+                      </div>
+                    </div>
+                    </center>
+                  </Modal>
```
Comment (joshdoman) [inline src/pages/Post.js:768]: Let's refactor the model into its own subcomponent. We can pass all the necessary functions into it as props.

**Review by joshdoman — CHANGES_REQUESTED** [review]
Looks great! I've added a few questions and made a few style suggestions above. This file is quickly becoming a monolith and needs to be broken up.

File: src/pages/Post.js (hunk 1 above)
Comment (matthewrh) [reply src/pages/Post.js:542]: See #9

File: src/pages/Post.js (hunk 3 above)
Comment (matthewrh) [reply src/pages/Post.js:552]: See #9

**Pull request merged**